botocore/utils.py

```diff
@@ -389,6 +389,23 @@ def parse_to_aware_datetime(value):
     return datetime_obj
 
 
+def datetime2timestamp(dt):
+    """Calculate the timestamp based on the given datetime instance.
+
+    Naive datetime input will be treated as UTC time rather than local time.
```

Comment:
Curious to see what others think about this.  A utility function further down in this module does the same thing, but that was specifically for back-compat to fix a regression.  Not sure if it makes sense to carry that forward.

We'd still have to support the no timezone == UTC in the CLI, but we don't necessarily have to have that throughout the botocore utility methods.  I could go either way.